src/Nethermind/Nethermind.Serialization.Rlp/BlockDecoder.cs

```diff
@@ -116,16 +116,14 @@ public class BlockDecoder : IRlpValueDecoder<Block>, IRlpStreamDecoder<Block>
 
                 if (lengthWasRead)
                 {
-                    int requestsLength = rlpStream.ReadSequenceLength();
-                    int requestsCheck = rlpStream.Position + requestsLength;
+                    // read an array of byte arrays
+                    byte[][] requestsEncoded = rlpStream.DecodeByteArrays();
```

Comment:
unneeded, make it lazy, call internal decoders based on type and progressing the stream